### ISSUE TITLE
Error on loading container details: invalid type string "SIGTERM", expected u64

### ISSUE THREAD

**Issue by Cloud0x1** [marhkb/pods#789]
The error appears when:
Opening "Containers" tab > Clicking on ANY kind of containers
I tried creating container from command-line, from the gui, tried different images, the error still appears each time consistently.

![image](https://github.com/marhkb/pods/assets/151971681/c9fdab52-eccd-49fa-8c99-78cef6689c2b)

```
$ cat /etc/os-release | head -1
NAME="Arch Linux"
$ grep -m 1 'model name' /proc/cpuinfo
model name	: Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz
```
Pods version: 2.0.1 (flatpak)

**Comment by katanaBadger** [comment]
I was in the process of submitting a bug report as well.

I have so far never came across an issue using Pods on Fedora Workstation, but after a recent Arch install, any container instance created with or without Pods always throws this toast message:

`"Error on loading container details: invalid type: string "SIGTERM", expected u64 at line 1 column xxxx"`
> Note: the column is a different 4 digit number with every container, usually in the 3700-4199 range.

Similarly, inspecting the container yields:

`"Inspection error: invalid type: string "SIGTERM", expected u64 at line 1 column xxxx"`

Due to this, the 'size' field is blank and loads indefinitely, and the 'Inspection' utility cannot be entered.

Also, I'm not sure if this is related in any way, but the only SIGTERM that appears in the container inspection is:

`"StopSignal": "SIGTERM",`

### Environment

- Pods version: 2.0.1 (flatpak)
- Podman version: 5.0.1
- OS: Arch Linux (kernel 6.8.4, GNOME 46)

**Comment by Cloud0x1** [comment]
Maybe this is 2.0.1 thing?

**Comment by marhkb** [comment]
It seems, Podman 5 had a breaking API change. So, serde can't deserialize the HTTP response. I'm thinking about to change to something like flexbuffers in order to handle such breakages easier. Also, https://github.com/vv9k/podman-api-rs seems to be unmaintained.

**Comment by woodsb02** [comment]
Looks like there are a few people requesting podman-api-rs gains support for the 5.0 api:
https://github.com/vv9k/podman-api-rs/issues/169

I'm dealing with another breaking change in the Podman 5.0 API, which is preventing pods from displaying containers correctly:
https://github.com/containers/podman/pull/13811

For me, this is causing the following error:

> Error on loading container details: invalid type: sequence, expected a string at line 1 column 5702

Running the following command:
`$ curl --unix-socket /run/user/1000/podman/podman.sock http://d/v4.3.1/libpod/containers/eebde641a1336aa7551c8384bf3dd2a2ac40296ec0ac1df543a2dead8ff38dd9/json?size=true`

I found that column 5702 was the colon in between:

> "Entrypoint":["docker-entrypoint.sh"]

![Screenshot from 2024-04-25 20-07-53](https://github.com/marhkb/pods/assets/7113557/1489d427-9304-4b3b-b038-d2b522aa3345)

**Comment by marhkb** [comment]
It seems the issue is fixed in podman 5.1.2.
